src/Microsoft.Extensions.Configuration.AzureAppConfiguration/JsonConfiguration/JsonConfigurationParser.cs

```diff
@@ -0,0 +1,115 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Mime;
+using System.Text.Json;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.JsonConfiguration
+{
+    internal static class JsonConfigurationParser
+    {
+        public static bool IsJsonContentType(string contentType)
+        {
+            if (string.IsNullOrWhiteSpace(contentType))
+            {
+                return false;
+            }
+            string acceptedMainType = "application";
+            string acceptedSubType = "json";
+
+            try
+            {
+                ContentType ct = new ContentType(contentType.Trim().ToLower());
+                var type = ct.MediaType;
+                if (type == FeatureManagementConstants.ContentType ||
+                    type == KeyVaultConstants.ContentType)
+                {
+                    return false;
+                }
+
+                if (type.Contains('/'))
+                {
+                    string mainType = type.Split('/')[0];
+                    string subType = type.Split('/')[1];
+                    if (mainType == acceptedMainType)
+                    {
+                        if (subType.Contains('+'))
+                        {
+                            var subTypes = subType.Split('+');
+                            if (Array.Exists(subTypes, x => x == acceptedSubType))
+                            {
+                                return true;
+                            }
+                        }
+                        else if (subType == acceptedSubType)
+                        {
+                            return true;
+                        }
+                    }
+                }
+            }
+            catch (Exception ex) when (ex is ArgumentException || ex is FormatException)
+            {
+                // not a valid JSON content type
+            }
+            return false;
+        }
+
+        public static List<KeyValuePair<string, string>> ParseJsonSetting(ConfigurationSetting setting)
+        {
+            List<KeyValuePair<string, string>> keyValues = new List<KeyValuePair<string, string>>();
+            Dictionary<string, string> keyValueDict = new Dictionary<string, string>();
+            ParseSetting(setting.Key, setting.Value, keyValueDict);
+            foreach (KeyValuePair<string, string> entry in keyValueDict)
+            {
+                keyValues.Add(entry);
+            }
+            return keyValues;
+        }
+
+
+        public static void ParseSetting(string currentKey, string currentValue, Dictionary<string, string> keyValueDict)
+        {
+            try
+            {
+                var json_data = JsonSerializer.Deserialize<JsonElement>(currentValue);
+                switch (json_data.ValueKind)
+                {
+                    case JsonValueKind.Array:
+                        var valueArray = json_data.EnumerateArray();
```

Comment:
>var [](start = 24, length = 3)

We can specify explicit type `ArrayEnumerator` to increase readability. Same for `ObjectEnumerator` type in the next case block.